[2025-05-02T17:14:15.749+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-05-02T17:14:15.780+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: weather_data_pipeline.collect_data manual__2025-05-02T17:07:35.124421+00:00 [queued]>
[2025-05-02T17:14:15.789+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: weather_data_pipeline.collect_data manual__2025-05-02T17:07:35.124421+00:00 [queued]>
[2025-05-02T17:14:15.790+0000] {taskinstance.py:2303} INFO - Starting attempt 2 of 2
[2025-05-02T17:14:15.808+0000] {taskinstance.py:2327} INFO - Executing <Task(BashOperator): collect_data> on 2025-05-02 17:07:35.124421+00:00
[2025-05-02T17:14:15.822+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=452) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-05-02T17:14:15.823+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'weather_data_pipeline', 'collect_data', 'manual__2025-05-02T17:07:35.124421+00:00', '--job-id', '69', '--raw', '--subdir', 'DAGS_FOLDER/weather_pipeline_dag.py', '--cfg-path', '/tmp/tmp7r4jhfnr']
[2025-05-02T17:14:15.826+0000] {standard_task_runner.py:63} INFO - Started process 454 to run task
[2025-05-02T17:14:15.827+0000] {standard_task_runner.py:91} INFO - Job 69: Subtask collect_data
[2025-05-02T17:14:15.888+0000] {task_command.py:426} INFO - Running <TaskInstance: weather_data_pipeline.collect_data manual__2025-05-02T17:07:35.124421+00:00 [running]> on host 0b0b6d30f653
[2025-05-02T17:14:15.992+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='weather_data_pipeline' AIRFLOW_CTX_TASK_ID='collect_data' AIRFLOW_CTX_EXECUTION_DATE='2025-05-02T17:07:35.124421+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-02T17:07:35.124421+00:00'
[2025-05-02T17:14:15.995+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-05-02T17:14:16.005+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-05-02T17:14:16.007+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', 'python /opt/***/scripts/collect_data.py']
[2025-05-02T17:14:16.025+0000] {subprocess.py:86} INFO - Output:
[2025-05-02T17:14:21.344+0000] {subprocess.py:93} INFO - API Response:  {'coord': {'lon': 67.0822, 'lat': 24.9056}, 'weather': [{'id': 802, 'main': 'Clouds', 'description': 'scattered clouds', 'icon': '03n'}], 'base': 'stations', 'main': {'temp': 28.11, 'feels_like': 31.75, 'temp_min': 28.11, 'temp_max': 28.11, 'pressure': 1005, 'humidity': 76, 'sea_level': 1005, 'grnd_level': 1002}, 'visibility': 10000, 'wind': {'speed': 6.24, 'deg': 237, 'gust': 6.47}, 'clouds': {'all': 36}, 'dt': 1746205345, 'sys': {'country': 'PK', 'sunrise': 1746147324, 'sunset': 1746194506}, 'timezone': 18000, 'id': 1174872, 'name': 'Karachi', 'cod': 200}
[2025-05-02T17:14:21.346+0000] {subprocess.py:93} INFO - Weather data saved successfully.
[2025-05-02T17:14:21.390+0000] {subprocess.py:97} INFO - Command exited with return code 0
[2025-05-02T17:14:21.393+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-05-02T17:14:21.439+0000] {taskinstance.py:1205} INFO - Marking task as SUCCESS. dag_id=weather_data_pipeline, task_id=collect_data, execution_date=20250502T170735, start_date=20250502T171415, end_date=20250502T171421
[2025-05-02T17:14:21.501+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-05-02T17:14:21.528+0000] {taskinstance.py:3482} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-05-02T17:14:21.532+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
